### PR TITLE
Fix no shadow rule

### PR DESCRIPTION
**Pull request by nikolaevtomov**
Fix no-shadow rule https://github.com/dabapps/eslint-config-dabapps/issues/74

The solution was found in the documentation on the `How to use` section. https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md#how-to-use

Note: I intended to publish this as `patch v7.0.1`

### CODE REVIEW

File: base/.eslintrc.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "no-self-compare": 2,
     "no-useless-call": 2,
     "no-shadow-restricted-names": 2,
-    "no-shadow": 2,
+    "no-shadow": 0,
+    "@typescript-eslint/no-shadow": 2,
```
Comment (JakeSidSmith) [inline base/.eslintrc.json:30]: This needs to go into the overrides in the typescript config.

**Review by JakeSidSmith — CHANGES_REQUESTED** [review]

File: base/.eslintrc.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
     "no-self-compare": 2,
     "no-useless-call": 2,
     "no-shadow-restricted-names": 2,
-    "no-shadow": 2,
+    "no-shadow": 0,
```
Comment (JakeSidSmith) [inline base/.eslintrc.json:29]: This also needs moving into the typescript overrides.

Comment (nikolaevtomov) [reply base/.eslintrc.json]: Done

File: base/.eslintrc.json [hunk 3]
```diff
@@ -26,7 +26,6 @@
     "no-self-compare": 2,
     "no-useless-call": 2,
     "no-shadow-restricted-names": 2,
-    "no-shadow": 2,
```
Comment (JakeSidSmith) [inline base/.eslintrc.json:29]: Put this back.

Comment (nikolaevtomov) [reply base/.eslintrc.json:29]: https://github.com/dabapps/eslint-config-dabapps/pull/75#discussion_r592247539

Comment (nikolaevtomov) [reply base/.eslintrc.json:29]: Did you mean to duplicate the rule but in the case file keep it as `"no-shadow": 0,`?

Comment (JakeSidSmith) [reply base/.eslintrc.json:29]: This error only occurs in typescript files, so you only need to adjust the rules in the typescript overrides. Instead you have removed the rule from our base config which means that none of our javascript files will be linted for this rule.

Comment (nikolaevtomov) [reply base/.eslintrc.json:29]: 👍

**Review by JakeSidSmith — APPROVED** [review]

**Pull request merged**